code/outils.ss

```diff
@@ -1,37 +1,75 @@
 ;; helpful procedures
 
+;; Convenience for grabbing alist values by key
 (define (lookup key alist)
   (cond ((assoc key alist) => cdr)
-	(else (error 'lookup "key not in alist" key alist))))
+        (else (error 'lookup "key not in alist" key alist))))
 
 (define (lookup-partial key)
   (lambda (alist)
     (lookup key alist)))
 
 (define (lookup-spine keys alist)
   (fold-left (lambda (alist key)
-	       (lookup key alist))
-	     alist
-	     keys))
+               (lookup key alist))
+             alist
+             keys))
 
 (define (read-all)
   (let loop ((sexp (read)) (config '()))
     (if (eof-object? sexp)
-	(reverse config)
-	(loop (read) (cons sexp config)))))
+        (reverse config)
+        (loop (read) (cons sexp config)))))
 
 (define (write-expression-to-file code file)
   (when (file-exists? file)
     (delete-file file))
   (with-output-to-file file
     (lambda ()
       (for-each (lambda (line)
-		  (pretty-print line) (newline))
-		code))))
+                  (pretty-print line) (newline))
+                code))))
 
 (define (update-config)
   (load "code/config.ss"))
 
+;; Begin Preprocessing/Helpers for make-config
+;;
+;; Top level helper for make-config
+(define (process-config)
+  (map (lambda (x)
+         (if (not (eq? (car x) 'exercises))
+             x
+             (cons 'exercises
+                   (exercises->snake-case
+                    (remp (lambda (exercise)
+                            (memq 'wip (map car exercise)))
+                          (cdr x))))))
+       track-config))
 
+;; Top level helper for process-config
+(define (exercises->snake-case exercises)
+  (let ((handle (lambda (exercise)
+                  (map format-js-pair exercise))))
+    (map handle exercises)))
 
+;; Helpers for exercises->snake-case
+(define (kebab->snake str)
+  (let ((k->s (lambda (c) (if (char=? c #\-) #\_ c))))
+    (apply string (map k->s (string->list str)))))
 
+(define (symbol->snake-case-string symbol)
+  (kebab->snake (symbol->string symbol)))
+
+(define (format-js-pair pair)
+  (let ((snake-key (symbol->snake-case-string (car pair))))
+    (if (null? (cdr pair))
+        `(,snake-key)
+        ;; sort the topics list
+        (if (symbol=? 'topics (car pair))
```

Comment:
`eq?`

Reply:
Would `eq?` be better here?  I figured the more specific `symbol=?` would be better.

Reply:
I think `symbol=?` is hardly ever used

Reply:
Okay.

Reply:
It's not. 0 times in `scmutils`. 1 time in the `srfi` collection. In `chezscheme` pretty much only used at the site of definition

Reply:
(according to grep)

Reply:
[https://www.scheme.com/tspl4/objects.html#./objects:h2](https://www.scheme.com/tspl4/objects.html#./objects:h2)

Indeed, `symbol=?` isn't even as efficient as `eq?`

Reply:
Done

Reply:
That is wild!

I wonder why it even exists then...